src/utils/work/getExtraDependencies.js

```diff
@@ -17,12 +17,17 @@ const getClusteringSettings = async (experimentId, body, dispatch, getState) =>
   return clusteringSettings;
 };
 
-// Check that the cell sets within the selected cellSetKey didn't change
+// Check that the cell sets within the selected selectedCellSet didn't change
 // e.g., if cell set was deleted we can't use cache
 const getSelectedCellSet = async (experimentId, body, dispatch, getState) => {
+  if (!body.downsampleSettings) return '';
+
   await dispatch(loadCellSets(experimentId));
 
-  const children = getCellSetsHierarchyByKeys([body.cellSetKey])(getState());
+  const [{ children }] = getCellSetsHierarchyByKeys(
+    [body.downsampleSettings.selectedCellSet],
+  )(getState());
+
   const cellSetsKeys = children.map((cellSet) => cellSet.key);
```

Comment:
in this function definition: `const getCellSets = async (experimentId, body, dispatch, getState) => {` `body` is not used. You can remove the parameter

Reply:
yes, but it is passed in the calls to `dependencyGetter`, so I need to add it simply to pick the next params.